src/EPR.Calculator.Frontend/wwwroot/js/upload.js

```diff
@@ -5,16 +5,32 @@
 
 function checkstatus() {
     $.ajax({
-        url: 'ProcessData',
-        type: 'POST',
-        success: function (response) {
-            window.location.href = response.newUrl;
+        url:'',
+        type: 'GET',
+        success: function (response,xhr) {          
+            callController(xhr.statusText, response)
         },
-        error: function () {
-            $window.location.href = response.newUrl;
+        error: function (response,xhr) {
+            callController(xhr.statusText, response)
         }
     })
 }
+function callController(type, data) {
+    var url = type === "success" ? "/ParameterConfirmation" : " /UploadCSVError";
+    $.ajax({
+            url: url,
+        type: 'POST',
+        contentType:"application/json",
+        data: JSON.stringify(data),
+        success: function () {           
+                window.location.href = url;
+            },
+            error: function (xhr, status, error) {
+                console.log(status, error);
```

Comment:
Do we need to integrate with Appinsights to handle errors? @johnbenjamin-defra  we need discuss it. Thanks

Reply:
@udaydefra - not sure, need to check what PI 1 team is doing about this. We can find it out and add something in our backlog to look after December 2024.

Reply:
Not sure we can wait on this till December. We need to raise it with Trilok